### PR TITLE
Add the minimum OS supported inside platform assemblies metadata

### PR DESCRIPTION
**Pull request by spouliot**
The minimum OS supported for .net assemblies are specified in
https://github.com/dotnet/core/blob/master/release-notes/6.0/6.0-supported-os.md

The minimum for legacy assemblies are unchanged. However this information
will now be present in the assemblies (and not just in documentation and
code).

This PR partially address https://github.com/xamarin/xamarin-macios/issues/10659 and possibly https://github.com/xamarin/xamarin-macios/issues/10728

Fix https://github.com/xamarin/xamarin-macios/issues/10793

### CODE REVIEW

File: Make.config [hunk 1]
```diff
@@ -189,6 +189,12 @@ MIN_WATCH_OS_VERSION=2.0
 MIN_TVOS_SDK_VERSION=9.0
 MIN_MACCATALYST_SDK_VERSION=13.1
 
+# Minimum OS versions for dotnet6 support
```
Comment (dalexsoto) [inline Make.config:192]: I wonder if we should have these inside the (Make.versions)[https://github.com/xamarin/xamarin-macios/blob/main/Make.versions] file. Just a nit really and possibly out of the scope of this PR

```shell
# Minimum OS versions for running XI/XM apps.
MIN_IOS_SDK_VERSION=7.0
MIN_OSX_SDK_VERSION=10.9
MIN_WATCHOS_SDK_VERSION=2.0
MIN_WATCHOS64_32_SDK_VERSION=5.1
MIN_WATCH_OS_VERSION=2.0
MIN_TVOS_SDK_VERSION=9.0
MIN_MACCATALYST_SDK_VERSION=13.1

# Minimum OS versions for dotnet6 support
DOTNET_MIN_IOS_SDK_VERSION=10.0
DOTNET_MIN_TVOS_SDK_VERSION=10.0
DOTNET_MIN_MACCATALYST_SDK_VERSION=13.1
DOTNET_MIN_OSX_SDK_VERSION=10.14
```

**Review by dalexsoto — APPROVED** [review]

File: tools/common/Make.common [hunk 2]
```diff
@@ -2,8 +2,18 @@
...
 		-e 's/@IOS_SDK_VERSION@/$(IOS_SDK_VERSION)/g' -e 's/@WATCHOS_SDK_VERSION@/$(WATCH_SDK_VERSION)/' -e 's/@TVOS_SDK_VERSION@/$(TVOS_SDK_VERSION)/' -e 's/@OSX_SDK_VERSION@/$(OSX_SDK_VERSION)/' \
 		-e 's/@MACCATALYST_SDK_VERSION@/$(MACCATALYST_SDK_VERSION)/g' \
-		-e 's/@MIN_IOS_SDK_VERSION@/$(MIN_IOS_SDK_VERSION)/g' -e 's/@MIN_WATCHOS_SDK_VERSION@/$(MIN_WATCHOS_SDK_VERSION)/' -e 's/@MIN_TVOS_SDK_VERSION@/$(MIN_TVOS_SDK_VERSION)/' -e 's/@MIN_OSX_SDK_VERSION@/$(MIN_OSX_SDK_VERSION)/' \
+		\
+		-e 's/@MIN_IOS_SDK_VERSION@/$(MIN_IOS_SDK_VERSION)/g' \
+		-e 's/@MIN_WATCHOS_SDK_VERSION@/$(MIN_WATCHOS_SDK_VERSION)/' \
+		-e 's/@MIN_TVOS_SDK_VERSION@/$(MIN_TVOS_SDK_VERSION)/' \
+		-e 's/@MIN_OSX_SDK_VERSION@/$(MIN_OSX_SDK_VERSION)/' \
 		-e 's/@MIN_MACCATALYST_SDK_VERSION@/$(MIN_MACCATALYST_SDK_VERSION)/g' \
+		\
```
Comment (rolfbjarne) [inline tools/common/Make.common:11]: Did you forget to push any changes to tools/common/SdkVersions.cs.in? Because these changes will only have any effect if that file is modified as well.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tools/common/Make.common (hunk 2 above)
Comment (spouliot) [reply tools/common/Make.common:11]: yes (good catch) and no :-)

I have local changes for the file, to expose the `DOTNET*` versions when `NET` is defined. I did not include it inside _this_PR since it was not used (and I could have split the above change across diff PR too).

One of the next PR will include it as it enables introspection to do more accurate testing (and detect attributes that are not required, when older than the minimum). Those will come with a lot of binding changes (to make the test pass).

**Review by rolfbjarne — APPROVED** [review]
